commit 6c584bc8337da0a19287d51eaf05c608711b9ef5
Author: Mark Nauwelaerts <mnauw@users.sourceforge.net>
Date:   Sun May 4 13:37:46 2014 +0200

    rtpjitterbuffer: avoid stall by corrupted seqnum accounting

https://bugzilla.gnome.org/show_bug.cgi?id=739709

Upstream Status: Backport

Signed-off-by: Song Bing <b06498@freescale.com>

diff --git a/gst/rtpmanager/gstrtpjitterbuffer.c b/gst/rtpmanager/gstrtpjitterbuffer.c
index 1459e91..62d7228 100644
--- a/gst/rtpmanager/gstrtpjitterbuffer.c
+++ b/gst/rtpmanager/gstrtpjitterbuffer.c
@@ -2826,7 +2826,10 @@ do_deadline_timeout (GstRtpJitterBuffer * jitterbuffer, TimerData * timer,
 
   GST_INFO_OBJECT (jitterbuffer, "got deadline timeout");
 
-  priv->next_seqnum = timer->seqnum;
+  /* timer seqnum might have been obsoleted by caps seqnum-base,
+   * only mess with current ongoing seqnum if still unknown */
+  if (priv->next_seqnum == -1)
+    priv->next_seqnum = timer->seqnum;
   remove_timer (jitterbuffer, timer);
   JBUF_SIGNAL_EVENT (priv);
 
